store/src/java-test/com/zimbra/cs/imap/ImapFolderTest.java

```diff
@@ -61,4 +61,47 @@ public void testGetSubsequence() throws Exception {
         Assert.assertNotNull(i4set);
         Assert.assertEquals(3, i4set.size());
     }
+
+    @Test
+    public void getSearchQuery() throws Exception {
+        ImapSearch seqSearch = new SequenceSearch("S", "1:*", true);
+        ImapCredentials creds = new ImapCredentials(acct, ImapCredentials.EnabledHack.NONE);
+        ImapPath path = new ImapPath("INBOX", creds);
+        byte params = 0;
+        ImapFolder i4folder = new ImapFolder(path, params, null);
+        for(int i = 1; i < (ImapFolder.LARGEST_FOLDER_BATCH * 3); i++) {
+            i4folder.cache(new ImapMessage(i, Type.of((byte) 5), i, 0, null), true);
+        }
+        String searchStr = i4folder.getSearchQuery(seqSearch);
+        Assert.assertEquals("wrong search query when sequence is 1:*", "in:\"INBOX\" item:all", searchStr);
+
+        //2X max batch size
+        seqSearch = new SequenceSearch("S", String.format("1:%d", ImapFolder.LARGEST_FOLDER_BATCH * 2), true);
+        searchStr = i4folder.getSearchQuery(seqSearch);
+        Assert.assertEquals("wrong search query when sequence is twice the size of max batch", "in:\"INBOX\" item:all", searchStr);
```

Comment:
This looks to me as if the search will return all items in the folder - but we only want the first 1200 items of 1800.
Am I missing something?
If there is something that cleans up the mess later, then perhaps a SOAP IMAP test would prove that is working.  Also, if that is the case, the code could definitely benefit from more comments to explain how it works - and this test should have a comment that the list of all items gets pruned later for the real IMAP case.